### PR TITLE
mysql: fix time parse

### PR DESCRIPTION
**Pull request by coocood**
Checks day for the particular month.

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @zimulala PTAL

**Comment by zimulala** [comment]
LGTM

File: mysql/time_test.go [hunk 1]
```diff
@@ -80,6 +80,8 @@ func (s *testTimeSuite) TestDateTime(c *C) {
 		"1000-01-01 00:00:70",
 		"1000-13-00 00:00:00",
 		"10000-01-01 00:00:00",
+		"1000-09-31 00:00:00",
+		"1001-02-29 00:00:00",
```
Comment (shenli) [inline mysql/time_test.go]: Should you consider leap year?

**Comment by coocood** [comment]
@shenli PTAL

**Comment by shenli** [comment]
LGTM

**Pull request merged**